pkg/kongstate/credentials.go

```diff
@@ -128,6 +133,18 @@ func NewOauth2Credential(config interface{}) (*Oauth2Credential, error) {
 	return &res, nil
 }
 
+func NewMTLSAuth(config interface{}) (*MTLSAuth, error) {
+	var res MTLSAuth
+	err := decodeCredential(config, &res.MTLSAuth)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode mTLS credential: %w", err)
```

Comment:
Codecov points out we have an untested failure condition here, is this feasible to add to our tests?

Reply:
904ff7a is a bit contrived, but hits that class of failures for mtls-auth and the rest.